### ISSUE TITLE
[BUG] pandas plotting backend not set before plot call

### ISSUE THREAD

**Issue by 8W9aG** [skfolio/skfolio#59]
**Describe the bug**

`skfolio` calls `pandas` `plot` to create its graphs. It sets the backend in places like [this](https://github.com/skfolio/skfolio/blob/67a2a18288fcecd5f911d67402e6839b0332fa5a/src/skfolio/portfolio/_base.py#L65) when `skfolio` is imported. However this may be changed by other operators, so when it comes time to plot issues like this can arise:

```
│ /Users/xxx/dev/xxx/venv/lib/python3.11/site-packages/skfolio/population/_population.py:6 │
│ 20 in plot_cumulative_returns                                                                    │
│                                                                                                  │
│   619 │   │   fig = df.plot()                                                                    │
│ ❱ 620 │   │   fig.update_layout(                                                                 │
│   621 │   │   │   title=title,                                                                   │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
AttributeError: 'Axes' object has no attribute 'update_layout'
```

In this case the backend has been changed to `matplotlib` by other libraries, while `skfolio` expects the backend to be `plotly`.

**To Reproduce**

```python
from skfolio.datasets import load_sp500_dataset
from skfolio.preprocessing import prices_to_returns
from skfolio.optimization import EqualWeighted
from skfolio import Population
import pandas as pd

prices = load_sp500_dataset()
X = prices_to_returns(prices)
model = EqualWeighted()
model.fit(X)
pop = Population([model.predict(X)])
pd.options.plotting.backend = "matplotlib"
pop.plot_cumulative_returns()
```

**Expected behavior**

No crash, during the `plot_cumulative_returns` call the backend is set to `plotly` and then set back again at the end of the call.

**Additional context**

I would be more than happy to make this change, just want to check that maintainers agree with the diagnosis.

**Versions**
0.2.1

**Comment by HugoDelatte** [comment]
Indeed! The solution would be to pass the backend in the plot function directly:
```python
fig = df.plot(backend="plotly")
```
With this solution, I'm wondering if it's good practice or not to keep this global setting?:
```python
pd.options.plotting.backend = "plotly"
```

**Comment by 8W9aG** [comment]
Your solution seems much better, I would agree with removing the global setting. Let me know if you want me to implement.

**Comment by HugoDelatte** [comment]
I'll take care of it as I need to ensure that all document plots are generated properly

**Comment by HugoDelatte** [comment]
Fixed in [v0.2.2](https://github.com/skfolio/skfolio/releases/tag/v0.2.2)
